### ISSUE TITLE
Global SnackBar Implementation

### ISSUE THREAD

**Issue by ajsarkar28** [department-of-veterans-affairs/va-mobile-app#9630]
# Project Status Overview

### Project Objective:

- The Design System team will be creating a new SnackBar component.  This Epic covers the Flagship Global team implementing that new component in the Global sections of the app.  
- The Flagship team will give feedback on the process to the Design System team to help them improve the Design System.

 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
|  |   |
|  |   |
|  |   |
</details>

### Monthly Updates

<summary>Updates</summary>

|Month | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| --------|---------|
| date | update

Implementation Notes:
some technical recommendations from tech discovery (may change during implementation):

- Remove bottomOffset from redux and generally remove the SnackBar (app level name, design system is Snackbar) component slice
- Design system Snackbar component will handle setting when showing Snackbar (since it is dismissed on navigation, should not need updating height)
- Should be able to clean up global.d.ts and src/App.tsx of custom config alongside plugging in the design system Snackbar per documentation
- While calling the Snackbar directly via snackbar.show(...) and snackbar.hideAll() will be possible, it is recommended to use design system provided convenience functions when possible as they bake in logic and automatically adjust to changes where manual calls may need more updating
- No longer true due to revisions during Snackbar creation such that now Snackbar manipulation is handled via the useSnackbar hook and not defined globally
- Remove SnackBarConstants from src/constants/common.ts as no longer relevant
- Recommend removal of SnackbarMessages typing that mandates success-/error-Msg properties and conditionally includes undo(Error)Msg
- Depends on screen, but rarely seems to be beneficial vs just plugging in the translation text--context both from surrounding logic and translation name clarifies what the Snackbar is appearing for
- Should be able to clean up appreciable snackbar-related parts of app-level theme

Additional note:

- When testing announcing the Snackbar for accessibility, some conflicting behavior with app-level setting focus to the back button was observed as documented on [this PR](https://github.com/department-of-veterans-affairs/va-mobile-library/pull/437) in secure messaging around moving a message to a different folder

Unclear how this should be resolved, but something to be cognizant during implementation into flagship and discuss with design system team if needed

- Snackbar accessibility functions correctly with RN 0.74 or higher. Flagship is currently on 0.73.

**Comment by oddballdave** [comment]
Duplicate of #9561

**Comment by kellylein** [comment]
@oddballdave #9561 is the parent Epic for this ticket, not a duplicate.

**Comment by oddballdave** [comment]
> [@oddballdave](https://github.com/oddballdave) [#9561](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9561) is the parent Epic for this ticket, not a duplicate.

My bad, I didn't see that sub-issue.

**Comment by rbontrager** [comment]
@oddballdave I'm seeing two issues

1. The issue discussed in the meeting where error snackbars are not flush with the bottom for fullScreenSubtasks.  I'm seeing this with the claims > upload files/upload photos screen, personal info > gender identity/preferred name, draft messages.
![Image](https://github.com/user-attachments/assets/4b3b1b93-0771-4d6c-bb65-e8aa3b1243ba)


2. I'm also seeing multiple instances where the error snackbars are not going away upon navigation.  This also looks to be affecting the fullScreenSubtasks screens. I can reproduce with the error snackbars on personal info > gender identity/preferred name and the error snackbar for messages > start new message if the message fails to send
https://github.com/user-attachments/assets/874406b9-fe65-4755-8270-f6e8babac49f

**Comment by jennb33** [comment]
12/9/2024 - this ticket is in PR; there were two issues: 
1) Disappearing Snackbar that was not dismissing the full-service modals
2) Snackbar is supposed to appear at different places on the screen

Dave is working on this right now; the existing codebase is also incorrect. He is looking for more guidance and is looking for the Figma layout for the Snackbar. Natasha will look into this to see if she can share it with Dave, specifically to the bottom nav.

**Comment by jennb33** [comment]
12/10/2024 - Dave has submitted a bunch of commits and has requested QA review on this ticket. 
He had to do another deep dive on the SnackBar component. There is a concept of a "safe area" and they differ between Android and iPhone. Referencing Figma and now has a way to get everything consistent (nav bar placement, etc). Forced errors.

**Comment by jennb33** [comment]
12/12/2024 - Dave made changes according to feedback. It seems like the SnackBar isn't going to be customizable. Tim approved PR, this still needs QA

**Comment by jennb33** [comment]
12/12/2024 @oddballdave we will use this same ticket for Sprint 7, for the QA work that still needs to happen.

**Comment by jennb33** [comment]
1/2/2025 - -sent to Don for QA approval

**Comment by jennb33** [comment]
1/6/2025- this spun @oddballdave off into a different direction. He's looking into other DS components. Dave had been patching issues, per the direction of the Core Mobile team. With Don's input, we are going to change our behavior there and won't be fixing by patches

**Comment by jennb33** [comment]
1/7/2025 - per @oddballdave this will be a Design System Library fix - it has to be released to the app stores, then will the fix can be done. Ticket https://github.com/department-of-veterans-affairs/va-mobile-feature-support/issues/152 is to determine the LOE for fixing the Design System library.

**Comment by jennb33** [comment]
1/8/2025 - this ticket could be blocking the potential solution for [9893](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9893). Once this ticket is complete, grab 9893 from the backlog and test to see if it fixed the issue.

**Comment by jennb33** [comment]
1/10/2025 - this ticket is rolling from Sprint 8 to Sprint 9, due to the library code update that will be done, rather than patching the fix.

**Comment by jennb33** [comment]
1/13/2025 - blocked pending Design System final version and code update solutions

**Comment by jennb33** [comment]
1/23/2025 - Rolling this ticket to Sprint 10, as it is still blocked, pending Design System final version and code update solutions

**Comment by jason-dehaan** [comment]
2/13/25 - Blocked until we get the design system updates integrated into the app (#192).

**Comment by jason-dehaan** [comment]
2-26-25 Matt is going to do a thorough, hands on QA, testing and verifying of this ticket.

**Comment by parkerbees** [comment]
Alex confirmed that this is in the 2.64 release. Moving to done.
